gapis/replay/mapping_exporter.go

```diff
@@ -30,74 +29,91 @@ import (
 	"github.com/google/gapid/gapis/api/transform"
 	"github.com/google/gapid/gapis/memory"
 	"github.com/google/gapid/gapis/replay/builder"
+	"github.com/google/gapid/gapis/replay/value"
 	"github.com/google/gapid/gapis/service"
 )
 
+type mappingHandle struct {
+	traceValue    uint64
+	replayAddress value.Pointer
+	size          uint64
+	name          string
+}
+
 type MappingExporter struct {
-	mappings *map[uint64][]service.VulkanHandleMappingItem
-	thread   uint64
-	path     string
-	results  map[uint64]service.VulkanHandleMappingItem
-	mutex    sync.RWMutex
+	mappings       *map[uint64][]service.VulkanHandleMappingItem
+	thread         uint64
+	path           string
+	traceValues    []mappingHandle
+	notificationID uint64
 }
 
 func NewMappingExporter(ctx context.Context, mappings *map[uint64][]service.VulkanHandleMappingItem) *MappingExporter {
 	return &MappingExporter{
-		mappings: mappings,
-		thread:   0,
-		path:     "",
-		results:  make(map[uint64]service.VulkanHandleMappingItem),
-		mutex:    sync.RWMutex{},
+		mappings:       mappings,
+		thread:         0,
+		path:           "",
+		traceValues:    make([]mappingHandle, 0, 0),
+		notificationID: 0,
 	}
 }
 
 func NewMappingExporterWithPrint(ctx context.Context, path string) *MappingExporter {
 	mapping := make(map[uint64][]service.VulkanHandleMappingItem)
 	return &MappingExporter{
-		mappings: &mapping,
-		thread:   0,
-		path:     path,
-		results:  make(map[uint64]service.VulkanHandleMappingItem),
-		mutex:    sync.RWMutex{},
+		mappings:       &mapping,
+		thread:         0,
+		path:           path,
+		traceValues:    make([]mappingHandle, 0, 0),
+		notificationID: 0,
 	}
 }
 
 func (m *MappingExporter) processNotification(ctx context.Context, s *api.GlobalState, n gapir.Notification) {
-	notificationData := n.GetData()
-	notificationID := n.GetId()
-	mappingData := notificationData.GetData()
-
-	m.mutex.Lock()
```

Comment:
You're removing a mutex here, and concurrency is never trivial: can you briefly document why in the commit message?